packages/common/log/src/processors/file-processor.ts

```diff
@@ -52,7 +54,31 @@ export const createFileProcessor = ({
       },
       context: jsonify(getContextFromEntry(entry)),
     };
-    appendFileSync(fd, JSON.stringify(record) + '\n');
+    let retryTS: number = 0;
+
+    // Retry writing if EAGAIN is encountered.
+    //
+    // Node may set stdout and stderr to non-blocking. https://github.com/nodejs/node/issues/42826
+    // This can cause EAGAIN errors when writing to them.
+    // In order to not drop logs, make log methods asynchronous, or deal with buffering/delayed writes, spin until write succeeds.
+
+    while (true) {
+      try {
+        return appendFileSync(fd, JSON.stringify(record) + '\n');
```

Comment:
I think a better solution would be to use an async version of this function. The logger doesn't need to wait for the log to be written to disk. The only caveat here is ensuring that the logs are written in the correct oreder